devtools/server/actors/replay/auth.js

```diff
@@ -336,10 +342,13 @@ Services.prefs.addObserver("devtools.recordreplay.user-token", () => {
 });
 
 // Init
-(() => {
+(async () => {
   initializeRecordingWebChannel();
   if (!hasOriginalApiKey()) {
     captureLastAuthId();
-    setReplayUserToken(validateUserToken());
+    // clear the token so we ensure that the token is communicated to connection.js
+    const token = await validateUserToken();
+    setReplayUserToken(null);
```

Comment:
Feeling like we're getting to the point where this file should export a function to add a listener, instead of forcing everything to use the pref observer.

Reply:
Yup. That's my thinking as well.